athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java

```diff
@@ -191,30 +182,41 @@ public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest getTabl
-                        final String partitionVal = BLOCK_PARTITION_COLUMN_NAME + "-limit-" + totalRecordCount + "-offset-" + offset;
-                        LOGGER.info("partitionVal {} ", partitionVal);
-                        blockWriter.writeRows((Block block, int rowNum) ->
-                        {
-                            block.setValue(BLOCK_PARTITION_COLUMN_NAME, rowNum, partitionVal);
-                            return 1;
-                        });
+                    // if number of partitions are more than defined limit "MAX_PARTITION_COUNT"
+                    // it will do maximum 50 partitions,49 partitions will have 500000 records each and last partition will have the remaining number of records.
+                    if (limit > SnowflakeConstants.MAX_PARTITION_COUNT) {
+                        for (int i = 1; i <= SnowflakeConstants.MAX_PARTITION_COUNT; i++) {
+                            int partitionRecord = SnowflakeConstants.PARTITION_RECORD_COUNT;
+                            if (i > 1) {
+                                offset = offset + SnowflakeConstants.PARTITION_RECORD_COUNT;
+                            }
+                            if (i == SnowflakeConstants.MAX_PARTITION_COUNT) {
+                                //Updating partitionRecord variable to display the remaining records in the last partition.
+                                //we get the value by subtracting the records displayed till 49th partition from the total number of records.
+                                partitionRecord = (int) totalRecordCount - (SnowflakeConstants.PARTITION_RECORD_COUNT * (SnowflakeConstants.MAX_PARTITION_COUNT - 1));
+                            }
+                            final String partitionVal = BLOCK_PARTITION_COLUMN_NAME + "-limit-" + partitionRecord + "-offset-" + offset;
+                            LOGGER.info("partitionVal {} ", partitionVal);
+                            blockWriter.writeRows((Block block, int rowNum) ->
+                            {
+                                block.setValue(BLOCK_PARTITION_COLUMN_NAME, rowNum, partitionVal);
+                                return 1;
+                            });
+                        }
                     }
                     else {
                         /**
                          * Custom pagination based partition logic will be applied with limit and offset clauses.
-                         * the partition values we are setting the limit and offste values like p-limit-3000-offset-0
+                         * the partition values we are setting the limit and offset values like p-limit-3000-offset-0
                          */
                         for (int i = 1; i <= limit; i++) {
                             if (i > 1) {
-                                offset = offset + totalpagecount;
+                                offset = offset + SnowflakeConstants.PARTITION_RECORD_COUNT;
```

Comment:
You can just static import these, no need to prefix them all the time.

Reply:
Hi Keshav,I have addressed all the review comments.Please review